vdr/store/memory.go

```diff
@@ -109,26 +109,41 @@ func (m *memory) Resolve(id did.DID, metadata *vdr.ResolveMetadata) (*did.Docume
 	}
 
 	// return a deep copy
-	doc := &did.Document{}
-	docMetadata := &vdr.DocumentMetadata{}
+	copyEntry, err := deepCopy(entry)
```

Comment:
not sure if we want to have deepCopy() return an error? A memory copy should alway be able to succeed imo. I believe @woutslakhorst checked why/when marshal/unmarshal can fail

Reply:
when implemented in go-did yes. Now it's just on par with what it was but a bit more separated.